From 55b93a38c023d7395df5ca2f6d22e436b350c9a3 Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Mon, 26 Jan 2009 15:19:47 -0600
Subject: [PATCH 221/304] OMAP3ISP: Backend: Better preview default values

Changed Contrast, Brightness, WB coefitioens default values.

Turned off: chroma suppress and luma enhance in preview engine default
configuration.

Signed-off-by: Gjorgji Rosikopulos <grosikopulos@mm-sol.com>
Signed-off-by: Sergio Aguirre <saaguirre@ti.com>
---
 drivers/media/video/isp/isppreview.c |   10 +++++-----
 drivers/media/video/isp/isppreview.h |    6 +++---
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/media/video/isp/isppreview.c b/drivers/media/video/isp/isppreview.c
index 8765752..c80c666 100644
--- a/drivers/media/video/isp/isppreview.c
+++ b/drivers/media/video/isp/isppreview.c
@@ -127,8 +127,8 @@ static struct ispprev_csc flr_prev_csc[] = {
 /* Default values in Office Flourescent Light for White Balance*/
 #define FLR_WBAL_DGAIN		0x100
 #define FLR_WBAL_COEF0		0x20
-#define FLR_WBAL_COEF1		0x35
-#define FLR_WBAL_COEF2		0x2b
+#define FLR_WBAL_COEF1		0x29
+#define FLR_WBAL_COEF2		0x2d
 #define FLR_WBAL_COEF3		0x20
 
 #define FLR_WBAL_COEF0_ES1	0x20
@@ -1934,14 +1934,14 @@ int __init isp_preview_init(void)
 	params->rgb2rgb = flr_rgb2rgb;
 	params->rgb2ycbcr = flr_prev_csc[ispprev_obj.color];
 
-	params->features = PREV_CFA | PREV_CHROMA_SUPPRESS | PREV_LUMA_ENHANCE
-				| PREV_DEFECT_COR | PREV_NOISE_FILTER;
+	params->features = PREV_CFA | PREV_DEFECT_COR | PREV_NOISE_FILTER;
 	params->features &= ~(PREV_AVERAGER | PREV_INVERSE_ALAW |
 						PREV_HORZ_MEDIAN_FILTER |
 						PREV_GAMMA_BYPASS |
 						PREV_DARK_FRAME_SUBTRACT |
 						PREV_LENS_SHADING |
-						PREV_DARK_FRAME_CAPTURE);
+						PREV_DARK_FRAME_CAPTURE |
+						PREV_CHROMA_SUPPRESS | PREV_LUMA_ENHANCE);
 	return 0;
 }
 
diff --git a/drivers/media/video/isp/isppreview.h b/drivers/media/video/isp/isppreview.h
index a49fe35..f8a12d5 100644
--- a/drivers/media/video/isp/isppreview.h
+++ b/drivers/media/video/isp/isppreview.h
@@ -26,16 +26,16 @@
 /* Isp query control structure */
 
 #define ISPPRV_BRIGHT_STEP		0x1
-#define ISPPRV_BRIGHT_DEF		0x1
+#define ISPPRV_BRIGHT_DEF		0x0
 #define ISPPRV_BRIGHT_LOW		0x0
 #define ISPPRV_BRIGHT_HIGH		0xF
 #define ISPPRV_BRIGHT_UNITS		0x7
 
 #define ISPPRV_CONTRAST_STEP		0x1
-#define ISPPRV_CONTRAST_DEF		0x2
+#define ISPPRV_CONTRAST_DEF		0x4
 #define ISPPRV_CONTRAST_LOW		0x0
 #define ISPPRV_CONTRAST_HIGH		0xF
-#define ISPPRV_CONTRAST_UNITS		0x5
+#define ISPPRV_CONTRAST_UNITS		0x4
 
 #define NO_AVE				0x0
 #define AVE_2_PIX			0x1
-- 
1.5.6

